test/react-native/features/steps/react-native-steps.rb

```diff
@@ -16,27 +16,36 @@
   }
 end
 
-Then('the app is not running') do
-  Maze::Wait.new(interval: 1, timeout: 10).until do
-    case Maze::Helper.get_current_platform
-    when 'ios'
-      state = Maze.driver.app_state('org.reactjs.native.example.reactnative')
-      $logger.info "The app is #{state}"
-      state == :not_running
-    when 'android'
-      state = Maze.driver.app_state('com.reactnative')
-      $logger.info "The app is #{state}"
-      # workaround for faulty app state detection in appium v1.23 and lower on
-      # Android where an app that is not running is detected to be running in
-      # the background
-      state == :not_running || state == :running_in_background
-    end
+# Waits for up to 10 seconds for the app to stop running.  It seems that Appium doesn't always
+# get the state correct (e.g. when backgrounding the app, or on old Android versions), so we
+# don't fail if it still says running after the time allowed.
+def wait_for_app_state(expected_state)
+  max_attempts = 20
+  attempts = 0
+  state = get_app_state
+  until (attempts >= max_attempts) || state == expected_state
+    attempts += 1
+    state = get_app_state
+    sleep 0.5
+  end
+  $logger.warn "App state #{state} instead of #{expected_state} after 10s" unless state == expected_state
+  state
```

Comment:
I think this is equivalent:

```suggestion
  wait = Maze::Wait.new(interval: 0.5, timeout: 10)
  state = get_app_state

  success = wait.until do
    state = get_app_state
    state == expected_state
  end

  $logger.warn "App state #{state} instead of #{expected_state} after 10s" unless success

  state
```

Reply:
Thanks - I shall try to remember to update this the next time I'm there, rather than suffer a full rebuild on CI right now.